### PR TITLE
non-split-shopping-list-items-endpoint-tests

### PR DESCRIPTION
**Pull request by shubham-khandelwal2-metacube**
## PR Description

https://spryker.atlassian.net/browse/CC-16843

### CODE REVIEW

File: tests/api/suite_nonsplit/glue/shopping_list_endpoints/shopping_list_items/negative.robot [hunk 1]
```diff
@@ -0,0 +1,451 @@
...
+    [Setup]    Run Keywords    I get access token for the customer:    ${yves_user.email}
+    ...    AND    I set Headers:    Content-Type=${default_header_content_type}    Authorization=${token}  
+    ...    AND    I send a POST request:    /shopping-lists    {"data":{"type":"shopping-lists","attributes":{"name":"${shopping_list_name}${random}"}}}
+    ...    AND    Response status code should be:    201
+    ...    AND    Save value to a variable:    [data][id]    shoppingListId
+    I send a POST request:    /shopping-lists/${shoppingListId}/shopping-list-items    {"data":{"type":"shopping-list-items","attributes":{"sku":"${concrete_available_product.sku}","quantity":99999999999999999999}}}
+    And Response status code should be:    422
+    And Response should return error code:    901
+    And Response reason should be:    Unprocessable Content
+    And Response should return error message:    quantity => This value should be of type integer.
```
Comment (Katerynaa-Lavryk) [inline tests/api/suite_nonsplit/glue/shopping_list_endpoints/shopping_list_items/negative.robot:106]: Another error message can also be checked here: "quantity => This value should be less than ... "

File: tests/api/suite_nonsplit/glue/shopping_list_endpoints/shopping_list_items/negative.robot [hunk 2]
```diff
@@ -0,0 +1,451 @@
...
+Add_an_abstract_product_to_the_shopping_list    
+    [Setup]    Run Keywords    I get access token for the customer:    ${yves_user.email}
+    ...    AND    I set Headers:    Content-Type=${default_header_content_type}    Authorization=${token}  
+    ...    AND    I send a POST request:    /shopping-lists    {"data":{"type":"shopping-lists","attributes":{"name":"${shopping_list_name}${random}"}}}
+    ...    AND    Response status code should be:    201
+    ...    AND    Save value to a variable:    [data][id]    shoppingListId
+    I send a POST request:    /shopping-lists/${shoppingListId}/shopping-list-items    {"data":{"type":"shopping-list-items","attributes":{"sku":"${bundle_product.abstract.sku}","quantity":1}}}
+    And Response status code should be:    422
+    And Response reason should be:    Unprocessable Content
+    And Response should return error message:    1508
```
Comment (Katerynaa-Lavryk) [inline tests/api/suite_nonsplit/glue/shopping_list_endpoints/shopping_list_items/negative.robot:120]: You can also add assertion for error message here - "Concrete product not found."

File: tests/api/suite_nonsplit/glue/shopping_list_endpoints/shopping_list_items/negative.robot [hunk 3]
```diff
@@ -0,0 +1,451 @@
...
+    [Setup]    Run Keywords    I get access token for the customer:    ${yves_user.email}
+    ...    AND    I set Headers:    Content-Type=${default_header_content_type}    Authorization=${token}  
+    ...    AND    I send a POST request:    /shopping-lists    {"data":{"type":"shopping-lists","attributes":{"name":"${shopping_list_name}${random}"}}}
+    ...    AND    Response status code should be:    201
+    ...    AND    Save value to a variable:    [data][id]    shoppingListId
+    I send a POST request:    /shopping-lists/${shoppingListId}/shopping-list-items    {"data":{"type":"shopping-list-items","attributes":{"sku":"${concrete_available_product.sku}","quantity":"test"}}}    
+    And Response status code should be:    422
+    And Response should return error code:    901
+    And Response reason should be:    Unprocessable Content
+    And Response should return error message:    quantity => This value should be of type integer.
```
Comment (Katerynaa-Lavryk) [inline tests/api/suite_nonsplit/glue/shopping_list_endpoints/shopping_list_items/negative.robot:181]: Another error message can also be checked here: "quantity => This value should be greater than 0."

File: tests/api/suite_nonsplit/glue/shopping_list_endpoints/shopping_list_items/negative.robot [hunk 4]
```diff
@@ -0,0 +1,451 @@
...
+    I send a POST request:    /shopping-lists/${1st_shopping_list.id}/shopping-list-items    {"data":{"type":"shopping-list-items","attributes":{"sku":"${concrete_available_product.sku}","quantity":1}}}
+    And Response status code should be:    403
+    And Response should return error code:    002
+    And Response reason should be:    Forbidden
+    And Response should return error message:    Missing access token.
+
+Add_a_concrete_product_to_the_shopping_list_with_wrong_access_token  
+    [Setup]    Run Keywords    I get access token for the customer:    ${yves_user.email}
+    ...    AND    I set Headers:    Content-Type=${default_header_content_type}    Authorization=${token}1
+    ...    AND    I send a POST request:    /shopping-lists    {"data":{"type":"shopping-lists","attributes":{"name":"${shopping_list_name}${random}"}}}
```
[the remark this answers is not in the document]
Comment (shubham-khandelwal2-metacube) [reply tests/api/suite_nonsplit/glue/shopping_list_endpoints/shopping_list_items/negative.robot:21]: FIXED

File: tests/api/suite_nonsplit/glue/shopping_list_endpoints/shopping_list_items/negative.robot [hunk 5]
```diff
@@ -0,0 +1,451 @@
...
+Add_a_product_with_non_existing_sku_to_the_shopping_list    
+    [Setup]    Run Keywords    I get access token for the customer:    ${yves_user.email}
+    ...    AND    I set Headers:    Content-Type=${default_header_content_type}    Authorization=${token}  
+    ...    AND    I send a POST request:    /shopping-lists    {"data":{"type":"shopping-lists","attributes":{"name":"${shopping_list_name}${random}"}}}
+    ...    AND    Response status code should be:    201
+    ...    AND    Save value to a variable:    [data][id]    shoppingListId
+    I send a POST request:    /shopping-lists/${shoppingListId}/shopping-list-items    {"data":{"type":"shopping-list-items","attributes":{"sku":"sku${random}","quantity":1}}}
+    And Response status code should be:    422
+    And Response reason should be:    Unprocessable Content
+    And Response should return error message:    1508
```
[the remark this answers is not in the document]
Comment (shubham-khandelwal2-metacube) [reply tests/api/suite_nonsplit/glue/shopping_list_endpoints/shopping_list_items/negative.robot:46]: FIXED. 
Getting error message "Concrete product not found." in "code" key.

File: tests/api/suite_nonsplit/glue/shopping_list_endpoints/shopping_list_items/negative.robot (hunk 1 above)
Comment (shubham-khandelwal2-metacube) [reply tests/api/suite_nonsplit/glue/shopping_list_endpoints/shopping_list_items/negative.robot:106]: FIXED

File: tests/api/suite_nonsplit/glue/shopping_list_endpoints/shopping_list_items/negative.robot (hunk 3 above)
Comment (shubham-khandelwal2-metacube) [reply tests/api/suite_nonsplit/glue/shopping_list_endpoints/shopping_list_items/negative.robot:181]: DONE

File: tests/api/suite_nonsplit/glue/shopping_list_endpoints/shopping_list_items/negative.robot (hunk 5 above)
[the remark this answers is not in the document]
Comment (Katerynaa-Lavryk) [reply tests/api/suite_nonsplit/glue/shopping_list_endpoints/shopping_list_items/negative.robot:51]: Please change the expected result to have correct response  {"errors":[{"code":"1508","status":422,"detail":"Concrete product not found."}]} and add bug ID to this test
https://spryker.atlassian.net/browse/CC-19379

[the remark this answers is not in the document]
Comment (shubham-khandelwal2-metacube) [reply tests/api/suite_nonsplit/glue/shopping_list_endpoints/shopping_list_items/negative.robot]: DONE

**Review by Katerynaa-Lavryk — APPROVED** [review]

**Pull request merged**